src/Network/Request.php

```diff
@@ -1375,6 +1381,42 @@ public function setInput($input)
     }
 
     /**
+     * Update the request with a new request data element.
+     *
+     * Returns an updated request object. This method returns
+     * a *new* request object and does not mutate the request in-place.
+     *
+     * @param string $name The dot separated path to insert $value at.
+     * @param mixed $value The value to insert into the request data.
+     * @return self
```

Comment:
Updated to `static`. I originally used `self` as diactoros uses `self` in their API docs.